File: fixtures/projects/e2e-app/src/api/v2/handler.ts
import type { ApiResponse } from "../../types/api";
import { HTTP_STATUS } from "../../constants/http";

export type Status = "pending" | "running" | "done" | "error";

export type ApiV2Context = {
	requestId: string;
	method: string;
	path: string;
	body?: unknown;
};

export function handleRequest(ctx: ApiV2Context): ApiResponse<unknown> {
	if (ctx.method === "GET") {
		return { success: true, data: { version: 2, path: ctx.path, id: ctx.requestId } };
	}
	if (ctx.method === "POST" && ctx.body) {
		return { success: true, data: ctx.body };
	}
	return { success: false, error: { code: HTTP_STATUS.BAD_REQUEST, message: "Unsupported v2 operation" } };
}

export function parseV2Path(raw: string): string[] {
	return raw.split("/").filter(Boolean);
}

File: fixtures/projects/e2e-python/src/api/v2/handler.py
from src.types.api import APIRequest, APIResponse


def _build_meta(path: str) -> dict[str, str]:
    return {"version": "v2", "resource": path.split("/")[-1] or "root"}


def handle_request(request: APIRequest) -> APIResponse:
    meta = _build_meta(request.path)
    body = {
        "version": meta["version"],
        "method": request.method,
        "resource": meta["resource"],
    }
    status = 202 if request.method == "POST" else 200
    return APIResponse(status=status, body=body, headers={"x-api-version": "v2"})
